ui/src/external/monaco.fluxCompletions.ts

```diff
@@ -1,20 +1,83 @@
+// Libraries
+import {completion, sendMessage} from 'src/external/monaco.lspMessages'
+import {
+  MonacoToProtocolConverter,
+  ProtocolToMonacoConverter,
+} from 'monaco-languageclient/lib/monaco-converter'
+import {get} from 'lodash'
+
 // Types
-import {MonacoType} from 'src/types'
+import {CompletionItem} from 'monaco-languageclient/lib/services'
+import {MonacoType, FLUXLANGID, Server} from 'src/types'
+import {IDisposable} from 'monaco-editor'
+
+const m2p = new MonacoToProtocolConverter()
+const p2m = new ProtocolToMonacoConverter()
+
+export const registerCompletion = (
+  monaco: MonacoType,
+  server: Server
+): IDisposable => {
+  const completionProvider = monaco.languages.registerCompletionItemProvider(
+    FLUXLANGID,
+    {
+      provideCompletionItems: (model, position, context) => {
+        const wordUntil = model.getWordUntilPosition(position)
+        const defaultRange = new monaco.Range(
+          position.lineNumber,
+          wordUntil.startColumn,
+          position.lineNumber,
```

Comment:
what if the range is across lines?

Reply:
I don't think we provide completion items for multi line selections? it only happens while user is typing. so I don't think it's a problem.